src/client/handlers/NodesGetAll.ts

```diff
@@ -26,25 +26,18 @@ class NodesGetAll extends ServerHandler<
   ): AsyncGenerator<ClientRPCResponseResult<NodesGetMessage>> {
     if (ctx.signal.aborted) throw ctx.signal.reason;
     const { nodeGraph, keyRing } = this.container;
-    for await (const bucket of nodeGraph.getBuckets()) {
-      let index;
-      for (const id of Object.keys(bucket)) {
-        const encodedId = nodesUtils.encodeNodeId(
-          IdInternal.fromString<NodeId>(id),
-        );
+    for await (const [index, bucket] of nodeGraph.getBuckets()) {
+      for (const [id, info] of bucket) {
+        const encodedId = nodesUtils.encodeNodeId(keyRing.getNodeId());
```

Comment:
We need to get the `NodeId` from the information we're iterating over. So use the `id` and not the `keyRing.getNodeId()` which would be our own node id..

Reply:
Oh, understood.

Reply:
This is now resolved.